### PR TITLE
Add throttling for a sending batch size [MAILPOET-3588]

### PR DESCRIPTION
**Pull request by lysyjan**
[MAILPOET-3588]

[MAILPOET-3588]: https://mailpoet.atlassian.net/browse/MAILPOET-3588

### CODE REVIEW

File: lib/Cron/Workers/SendingQueue/SendingThrottlingHandler.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  }
+
+  public function processSuccess(): void {
+    $throttlingSettings = $this->loadSettings();
+    if (!isset($throttlingSettings['batch_size'])) {
+      return;
+    }
+    $throttlingSettings['success_in_row'] = isset($throttlingSettings['success_in_row']) ? ++$throttlingSettings['success_in_row'] : 1;
+    $this->logger->info("MailPoet throttling: increase success_in_row to: {$throttlingSettings['success_in_row']}");
+    if ($throttlingSettings['success_in_row'] === self::SUCCESS_THRESHOLD_TO_INCREASE) {
```
Comment (costasovo) [inline lib/Cron/Workers/SendingQueue/SendingThrottlingHandler.php:65]: I would rather use `>=`. Right now === seems to be safe but we may introduce some changes in the future combined with the fact cron may run by accident in parallel and if we for some reason get over SUCCESS_THRESHOLD_TO_INCREASE we would never return to a bigger batch.

File: tests/integration/Cron/Workers/SendingQueue/SendingThrottlingHandlerTest.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace MailPoet\Test\Cron\Workers;
+
+use MailPoet\Cron\Workers\SendingQueue\SendingThrottlingHandler;
+use MailPoet\Settings\SettingsController;
+
+class SendingThrottlingHandlerTest extends \MailPoetTest {
```
Comment (costasovo) [inline tests/integration/Cron/Workers/SendingQueue/SendingThrottlingHandlerTest.php:8]: I think we should add test cases for increasing batch size after certain amount of successes and maybe also that the lowest possible batch size is 1.

**Review by costasovo — CHANGES_REQUESTED** [review]
Good job! I have just a few remarks.

**Review by costasovo — APPROVED** [review]
👍

**Review by veljkho — APPROVED** [review]

**Pull request merged**